gloo/test/base_test.h

```diff
@@ -54,19 +59,26 @@ class Barrier {
 
 enum Transport {
   TCP,
+  TCP_TLS,
```

Comment:
do we need a `#if` guard for this as well?

Reply:
instead I decided to remove #if guards from this file to make it consistent with TCP/UV transports